mail/mailer_test.go

```diff
@@ -74,17 +73,26 @@ func expect(t *testing.T, buf *bufio.Reader, expected string) error {
 	return nil
 }
 
-type connHandler func(int, *testing.T, net.Conn)
+type connHandler func(int, *testing.T, net.Conn, *net.TCPConn)
 
-func listenForever(l net.Listener, t *testing.T, handler connHandler) {
+func listenForever(l *net.TCPListener, t *testing.T, handler connHandler) {
 	connID := 0
 	for {
-		conn, err := l.Accept()
+		tcpConn, err := l.AcceptTCP()
 		if err != nil {
 			return
 		}
+
+		keyPair, err := tls.LoadX509KeyPair("../test/mail-test-srv/localhost/cert.pem", "../test/mail-test-srv/localhost/key.pem")
+		if err != nil {
+			t.Fatalf("loading keypair: %s", err)
```

Comment:
This introduces another instance of that "don't Fatalf from a goroutine" problem we were just discussing.

A simple fix would be to `t.Error` here, and then return from the goroutine. We should also fix line 83 to print the error.

This actually makes me realize `listenForever` is a bit of an anti-pattern: each of our tests that calls it will leak the goroutine. Not really a big deal because the goroutines all die at the end of the tests, but it means we don't have a nice clear synchronization point.